### DISCUSSION TITLE
SDK - extending Auth class partially

### DISCUSSION THREAD

**Discussion by kadlinobit** [directus/directus#16339] [Archive]
Hi, 
I am trying to find a way how to tweak the `refresh` function in Directus SDK Auth module. 
**My intention is to do an action in my frontend when refresh request fails (delete user data from local storage and set logged-in flag to false).**

So I would like to extend Auth class, while keeping all it's methods untouched and only customizing the `refresh` function, in which I would be calling the **parent** `refresh` function, such as: 
```js
async function refresh() {
   const result = await super.refresh()
   
   if(result === 'error') {
      /// remove userdata and set logged-in to false
   }
   return result
}
```
[SDK docs](https://docs.directus.io/reference/sdk.html#auth) only mention extending the IAuth class, but that requires implementing all the methods by ourselves from scratch. 

Anyone has an idea how to extend Auth class to be able to call the parent methods while adding some custom wrapping code?

**Comment by rijkvanzanten** [comment]
I don't think there's a way to achieve that currently. I'd love to learn some about how you think it should work ideally though!

**Reply by kadlinobit** [reply]
Hi, thanks for the answer. To your question - I have no idea how this should work, as I have never used a javascript class based style of programming. 

But anyway, I guess that my use-case (checking refresh request result to be able to decide whether user is still logged in) might have a solution in axios interceptor - just found it in [this discussion](https://github.com/directus/sdk/discussions/37).


I'll dig into that.
